python/sglang/srt/models/mllama4.py

```diff
@@ -110,13 +112,25 @@ def permute(w: torch.Tensor, n_heads: int):
 
         # rotary embeds should be sliced
         if ("wk" in modules or "k_proj" in modules) and modules[-1] == "weight":
-            loaded_weight = permute(
-                loaded_weight, self.language_model.config.num_key_value_heads
-            )
+            if _is_cpu:
+                dim = getattr(
```

Comment:
There is a potential risk of these code spreading to various models.

And could we make `original_total_num_kv_heads` must be available when `_is_cpu == True` ?